src/librustc/mir/mod.rs

```diff
@@ -1988,7 +1988,7 @@ impl<'tcx> Operand<'tcx> {
     pub fn to_copy(&self) -> Self {
         match *self {
             Operand::Copy(_) | Operand::Constant(_) => self.clone(),
-            Operand::Move(ref place) => Operand::Copy(place.clone()),
+            Operand::Move(ref place) => Operand::Copy(*place),
```

Comment:
I wonder if there is a lint for `ref x` followed by `*x` - because you can just remove the `ref` and the `*`.